cloud/deployment/workerqueue/workerqueue.go

```diff
@@ -0,0 +1,165 @@
+package workerqueue
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/astronomer/astro-cli/astro-client"
+	"github.com/astronomer/astro-cli/cloud/deployment"
+	"github.com/pkg/errors"
+)
+
+var (
+	ErrWorkerQueueDefaultOptions = errors.New("failed to get worker-queue default options")
+	ErrInvalidWorkerQueueOption  = errors.New("worker-queue option is invalid")
+	ErrCannotUpdateExistingQueue = errors.New("worker-queue exists")
+)
+
+func Create(ws, deploymentID, name string, isDefaultWQueue bool, wQueueMin, wQueueMax, wQueueConcurrency int, client astro.Client, out io.Writer) error {
+	var (
+		requestedDeployment       astro.Deployment
+		err                       error
+		errHelp                   string
+		workerQueueToCreate       *astro.WorkerQueue
+		wQueueListToCreate        []astro.WorkerQueue
+		workerQueueDefaultOptions astro.WorkerQueueDefaultOptions
+		pool                      astro.NodePool
+		nodePoolIDForWorkerQueue  string
+	)
+	// get or select the deployment
+	requestedDeployment, err = deployment.GetDeployment(ws, deploymentID, client)
+	if err != nil {
+		return err
+	}
+
+	// TODO should we only get defaults once via init()
+	// get defaults for min-count, max-count and concurrency from API
+	workerQueueDefaultOptions, err = GetWorkerQueueDefaultOptions(client)
+	if err != nil {
+		return err
+	}
+
+	// Get the default nodepool ID to use
+	for _, pool = range requestedDeployment.Cluster.NodePools {
+		if pool.IsDefault {
+			nodePoolIDForWorkerQueue = pool.ID
+		}
+	}
+
+	// TODO user selects nodePoolID for creating a queue on non-default nodepools
+
+	workerQueueToCreate = &astro.WorkerQueue{
+		Name:       name,
+		IsDefault:  isDefaultWQueue,
+		NodePoolID: nodePoolIDForWorkerQueue,
+	}
+	workerQueueToCreate = setWorkerQueueValues(wQueueMin, wQueueMax, wQueueConcurrency, workerQueueToCreate, workerQueueDefaultOptions)
+
+	err = IsWorkerQueueInputValid(workerQueueToCreate, workerQueueDefaultOptions)
+	if err != nil {
+		return err
+	}
+
+	if QueueExists(requestedDeployment.WorkerQueues, workerQueueToCreate) {
+		// create does not allow updating existing queues
+		errHelp = fmt.Sprintf("use worker-queue update %s instead", workerQueueToCreate.Name)
```

Comment:
same comment here on the `-`, will omit further comments in the review regarding this but might be worth a sweep changing the notation across the PR!